code/game/objects/items/glassjar.dm

```diff
@@ -31,75 +33,109 @@
 /obj/item/glass_jar/MouseDrop(atom/target)
 	if(ishuman(target) || issmall(target) && target == usr && use_check_and_message(usr) && Adjacent(usr))
 		if(contains == JAR_GUMBALL)
-			handle_gumball_removal(usr)
+			if(length(gumballs_contained))
```

Comment:
gumballs should be able to be removed by attack_self like everything else, this is clunky